Feature/contacts

bortoletti-giorgia opened the pull request:
Hi, I created this branch for contact management.
A `Contact` is similar to a contact in the SIM so it is associated with a phone number and a name.
I saw @RicDeZen 's code for the events database and I decided to create a converter to switch from SMSPeer to Contact.
The manager `ContactManager` delegates the conversion at `ContactConverter`.
I also did two activities to view the saved contacts and add new ones. Editing and deletion are missing.

I should have written and explained everything in the specs.
Please, tell me if something is missing, if some part is not clear or if the architecture is wrong.
Thanks for the time you will use!

bortoletti-giorgia committed:
Added first implementation of Contact and its storage in a database

bortoletti-giorgia committed:
Added some tests

Added tests for ContactManager and ContactDatabase.

bortoletti-giorgia committed:
Added activities to add new contact and views all contacts

Contacts are inserted into the database through ContactManager.
Now we can insert new contact and view all saved contacts.

bortoletti-giorgia committed:
Changed the name of some attributes

turcato-niccolo commented on app/src/main/java/ingsw/group1/findmyphone/contacts/Contact.java:
```diff
@@ -0,0 +1,87 @@
+        if (name.equals(contactMaybeEquals.getName())
+            && address.equals(contactMaybeEquals.getAddress()))
+            return true;
+        //Should not reach this return
```
It should if the two contacts have different fields.

bortoletti-giorgia replied:
Like this:
`if (!name.equals(contactMaybeEquals.getName()) || !address.equals(contactMaybeEquals.getAddress()))
 return false;`
?

turcato-niccolo replied:
You don't need another test, returning false at this point is correct, because you're confronting two non-null contacts that have different fields. It's just the comment that is inapposite.

bortoletti-giorgia replied:
Ok, deleted.

turcato-niccolo commented:
Did you consider having a generic Contact<P>, and creating a Contact<SMSPeer>? Instead of storing a string for the address and creating a Parser I would do it like this, also because, to me, these contacts look a little fragile (`probabilmente sono mie pare`). 
Anyway, the contact manager can only add and remove SMSPeer, why not Add(Contact) and Remove(Contact)? Maybe it's because you're worried they could contain whatever string in the address?

bortoletti-giorgia commented:
I wanted to mask the creation of a `Contact` entity by necessarily passing through an SMSPeer that is what is taken from the UI. Also because as you said, we should repeat the checks on the addresses when instead passing an SMSPeer the checks are made by the constructor and we are sure that address is valid.
What do you think about this?
Do you have any suggestions to make a contact less fragile?

ricdezen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactDao.java:
```diff
@@ -0,0 +1,28 @@
+    @Override
+    public String getTableName(){
+        return DEFAULT_TABLE_NAME;
+    }
```
I believe count and get all would not work if the name is defined here, the table name should be in the Entity itself, otherwise the class name is used as a default

bortoletti-giorgia replied:
The tests are passed with count and getAll.
In the contact the table name is given from ContactDao as
`@Entity(tableName = ContactDao.DEFAULT_TABLE_NAME)`

How do you suggest to change?

ricdezen replied:
I'd just swap them, the constant in the Entity class and importing that constant in the Dao

ricdezen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactConverter.java:
```diff
@@ -0,0 +1,69 @@
+     * @return the String name of the {@link Contact} contactToConvert
+     */
+    @TypeConverter
+    public static String nameFromContact(Contact contactToConvert) {
```
Is this really necessary? We're just using a method to get the name of a Contact?

bortoletti-giorgia replied:
Right now no but in the contact list you see the name and not the number.
Only a list of names without addresses could be useful.

ricdezen replied:
Well, yes, but that's already built into the `Contact` class, that is why I'm asking what the purpose of this method is

RicDeZen commented:
To solve the fragility an SMSPeer should be used in the constructor, and a Room `Converter` should be used to tell the database how to convert an SMSPeer from and to a String

turcato-niccolo commented:
> I wanted to mask the creation of a `Contact` entity by necessarily passing through an SMSPeer that is what is taken from the UI. Also because as you said, we should repeat the checks on the addresses when instead passing an SMSPeer the checks are made by the constructor and we are sure that address is valid.

That I understand

> Do you have any suggestions to make a contact less fragile?

The first comment wasn't quite readable, I was suggesting a generic `Contact<P>`, and creating a `Contact<SMSPeer>` and as @RicDeZen said a constructor that takes a SMSPeer

RicDeZen commented:
Yes Contact as an Interface sounds good to me

bortoletti-giorgia commented:
Not bad. I'm working on it today.

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 25
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactManager.java  3
- app/src/main/java/ingsw/group1/findmyphone/contacts/Contact.java  6
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/ContactDatabaseTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/BaseDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactConverter.java  1
- app/src/test/java/ingsw/group1/findmyphone/contacts/ContactConverterTest.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/ContactManagerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954&bid=16307297)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java:
```diff
@@ -0,0 +1,57 @@
+
+    private RecyclerView recyclerView;
+    private RecyclerView.Adapter recyclerAdapter;
+    private FloatingActionButton newContactButton;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'newContactButton' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactDao.java:
```diff
@@ -0,0 +1,28 @@
+import androidx.room.Dao;
+import androidx.room.Ignore;
+
+import ingsw.group1.msglibrary.SMSMessage;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'ingsw.group1.msglibrary.SMSMessage'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - ingsw.group1.msglibrary.SMSMessage.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/Contact.java:
```diff
@@ -0,0 +1,87 @@
+import androidx.room.Ignore;
+import androidx.room.PrimaryKey;
+
+import java.util.zip.CheckedOutputStream;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - java.util.zip.CheckedOutputStream.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'java.util.zip.CheckedOutputStream'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/Contact.java:
```diff
@@ -0,0 +1,87 @@
+        if (obj == this)
+            return true;
+        Contact contactMaybeEquals = (Contact) obj;
+        if (name.equals(contactMaybeEquals.getName())
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/androidTest/java/ingsw/group1/findmyphone/contacts/ContactManagerTest.java:
```diff
@@ -0,0 +1,87 @@
+import java.io.IOException;
+
+import ingsw.group1.msglibrary.SMSPeer;
+import ingsw.group1.msglibrary.database.SMSDatabase;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - ingsw.group1.msglibrary.database.SMSDatabase.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'ingsw.group1.msglibrary.database.SMSDatabase'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/androidTest/java/ingsw/group1/findmyphone/contacts/ContactManagerTest.java:
```diff
@@ -0,0 +1,87 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - java.io.IOException.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'java.io.IOException'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/androidTest/java/ingsw/group1/findmyphone/contacts/ContactDatabaseTest.java:
```diff
@@ -0,0 +1,59 @@
+import org.junit.Test;
+
+import java.io.IOException;
+import static org.junit.Assert.*;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/test/java/ingsw/group1/findmyphone/contacts/ContactConverterTest.java:
```diff
@@ -0,0 +1,69 @@
+
+import ingsw.group1.msglibrary.SMSPeer;
+
+import static org.junit.Assert.*;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/test/java/ingsw/group1/findmyphone/contacts/ContactConverterTest.java:
```diff
@@ -0,0 +1,69 @@
+     * Conversion from a SMSPeer to a Contact with a name
+     */
+    @Test
+    public void contactFromSMSPeer_withValidName() {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'contactFromSMSPeer_withValidName' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/androidTest/java/ingsw/group1/findmyphone/contacts/ContactManagerTest.java:
```diff
@@ -0,0 +1,87 @@
+
+import android.content.Context;
+
+import androidx.room.Room;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - androidx.room.Room.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'androidx.room.Room'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java:
```diff
@@ -0,0 +1,52 @@
+
+    private EditText contactName;
+    private EditText contactPhone;
+    private Button newContactButton;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'newContactButton' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/MainActivity.java:
```diff
@@ -40,6 +42,7 @@
     private Button sendButton;
     private Button sendAlarmRequestButton;
     private Button sendLocationRequestButton;
+    private Button viewContacts;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'viewContacts' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'viewContacts' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/test/java/ingsw/group1/findmyphone/contacts/ContactConverterTest.java:
```diff
@@ -0,0 +1,69 @@
+    private static final String CONTACT_VALID_NAME = "NewContact";
+    private static final String CONTACT_EMPTY_NAME = "";
+
+    private Contact contactWithoutName, contactWithName; //a contact may or may not have a name
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactConverter.java:
```diff
@@ -0,0 +1,69 @@
+ *
+ * @author Giorgia Bortoletti
+ */
+class ContactConverter {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The utility class name 'ContactConverter' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java:
```diff
@@ -0,0 +1,57 @@
+ */
+public class ContactListActivity extends AppCompatActivity {
+
+    private RecyclerView recyclerView;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'recyclerView' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/androidTest/java/ingsw/group1/findmyphone/contacts/ContactManagerTest.java:
```diff
@@ -0,0 +1,87 @@
+import androidx.room.Room;
+import androidx.test.core.app.ApplicationProvider;
+
+import org.junit.After;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - org.junit.After.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.junit.After'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/test/java/ingsw/group1/findmyphone/contacts/ContactConverterTest.java:
```diff
@@ -0,0 +1,69 @@
+     * Conversion from a SMSPeer to a Contact without a name
+     */
+    @Test
+    public void contactFromSMSPeer_withEmptyName() {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'contactFromSMSPeer_withEmptyName' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java:
```diff
@@ -0,0 +1,57 @@
+public class ContactListActivity extends AppCompatActivity {
+
+    private RecyclerView recyclerView;
+    private RecyclerView.Adapter recyclerAdapter;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'recyclerAdapter' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/androidTest/java/ingsw/group1/findmyphone/contacts/ContactManagerTest.java:
```diff
@@ -0,0 +1,87 @@
+import ingsw.group1.msglibrary.SMSPeer;
+import ingsw.group1.msglibrary.database.SMSDatabase;
+
+import static org.junit.Assert.*;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

bortoletti-giorgia committed:
Added an abstract class GenericContact for Contact entity

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 27
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactAdapter.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/ContactDatabaseTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/Contact.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/GenericContact.java  6
- app/src/main/java/ingsw/group1/findmyphone/contacts/BaseDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactManager.java  3
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/ContactManagerTest.java  1
- app/src/test/java/ingsw/group1/findmyphone/contacts/ContactConverterTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactConverter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954&bid=16307297)

ricdezen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/GenericContact.java:
```diff
@@ -0,0 +1,99 @@
+ *
+ * @author Giorgia Bortoletti
+ */
+public abstract class GenericContact<P extends Peer<String>> {
```
I'd use an Interface personally, multiple interfaces can be implemented, but only one class can be extended, so, if it's not strictly necessary the logic should all be inside the implementing class.
If you agree with what I said above the interface could become `GenericContact<A, P extends Peer<A>>`

turcato-niccolo replied:
We tried to use abstract classes for kademlia's nodes and we ended up switching to interfaces, this situation is similar in my mind

ricdezen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/Contact.java:
```diff
@@ -16,72 +16,29 @@
-    @NonNull
-    @ColumnInfo(name = NAME_COLUMN_NAME)
-    private String name; //name: contact name
+public class Contact extends GenericContact<SMSPeer> {
```
I suggest maybe `SMSContact` as a name for the class to be a bit more clear

bortoletti-giorgia committed:
Changed GenericContact to an interface

- Changed name from Contact to SMSContact, and so for all the classes that had in their name 'Contact'
- Changed GenericContact from an abstract class to an interface
- Removed method ContactConverter#nameFromContact(Contact contactToConvert)
- Moved attribute for table name from ContactDao to SMSContact

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 15
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  1
- app/src/test/java/ingsw/group1/findmyphone/contacts/ContactConverterTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContact.java  6
- app/src/main/java/ingsw/group1/findmyphone/contacts/BaseDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactDatabaseTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactConverter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactDao.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactManagerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954&bid=16307297)

RicDeZen commented on app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactDatabaseTest.java:
```diff
@@ -0,0 +1,63 @@
+
+import ingsw.group1.msglibrary.SMSPeer;
+
+import static org.junit.Assert.*;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java:
```diff
@@ -0,0 +1,57 @@
+ */
+public class ContactListActivity extends AppCompatActivity {
+
+    private RecyclerView recyclerView;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'recyclerView' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'recyclerView' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/test/java/ingsw/group1/findmyphone/contacts/ContactConverterTest.java:
```diff
@@ -0,0 +1,69 @@
+    private static final String CONTACT_VALID_NAME = "NewContact";
+    private static final String CONTACT_EMPTY_NAME = "";
+
+    private SMSContact contactWithoutName, contactWithName; //a contact may or may not have a name
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java:
```diff
@@ -0,0 +1,57 @@
+public class ContactListActivity extends AppCompatActivity {
+
+    private RecyclerView recyclerView;
+    private RecyclerView.Adapter recyclerAdapter;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'recyclerAdapter' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'recyclerAdapter' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactManagerTest.java:
```diff
@@ -0,0 +1,82 @@
+
+import ingsw.group1.msglibrary.SMSPeer;
+
+import static org.junit.Assert.*;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java:
```diff
@@ -0,0 +1,52 @@
+
+    private EditText contactName;
+    private EditText contactPhone;
+    private Button newContactButton;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'newContactButton' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'newContactButton' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/test/java/ingsw/group1/findmyphone/contacts/ContactConverterTest.java:
```diff
@@ -0,0 +1,69 @@
+     * Conversion from a SMSPeer to a Contact without a name
+     */
+    @Test
+    public void contactFromSMSPeer_withEmptyName() {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'contactFromSMSPeer_withEmptyName' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'contactFromSMSPeer_withEmptyName' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java:
```diff
@@ -0,0 +1,57 @@
+
+    private RecyclerView recyclerView;
+    private RecyclerView.Adapter recyclerAdapter;
+    private FloatingActionButton newContactButton;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'newContactButton' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'newContactButton' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactDao.java:
```diff
@@ -0,0 +1,23 @@
+package ingsw.group1.findmyphone.contacts;
+
+import androidx.room.Dao;
+import androidx.room.Ignore;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - androidx.room.Ignore.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'androidx.room.Ignore'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'androidx.room.Ignore'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - androidx.room.Ignore.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/test/java/ingsw/group1/findmyphone/contacts/ContactConverterTest.java:
```diff
@@ -0,0 +1,69 @@
+     * Conversion from a SMSPeer to a Contact with a name
+     */
+    @Test
+    public void contactFromSMSPeer_withValidName() {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'contactFromSMSPeer_withValidName' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'contactFromSMSPeer_withValidName' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContact.java:
```diff
@@ -0,0 +1,102 @@
+        if (obj == this)
+            return true;
+        SMSContact contactMaybeEquals = (SMSContact) obj;
+        if (name.equals(contactMaybeEquals.getName())
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactConverter.java:
```diff
@@ -0,0 +1,57 @@
+ *
+ * @author Giorgia Bortoletti
+ */
+class SMSContactConverter {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The utility class name 'SMSContactConverter' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The utility class name 'SMSContactConverter' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The utility class name 'SMSContactConverter' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The utility class name 'SMSContactConverter' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

bortoletti-giorgia committed:
Merge branch 'development' into feature/contacts

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 15
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContact.java  6
- app/src/test/java/ingsw/group1/findmyphone/contacts/ContactConverterTest.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactManagerTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactDatabaseTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/BaseDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactConverter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactDao.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954&bid=16307297)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/MainActivity.java:
```diff
@@ -38,6 +39,7 @@
     private Button sendButton;
     private Button sendAlarmRequestButton;
     private Button sendLocationRequestButton;
+    private Button viewContacts;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'viewContacts' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'viewContacts' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

bortoletti-giorgia committed:
Changes after Codacy's suggestion

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 3
           

Complexity increasing per file
==============================
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactManagerTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/BaseDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactDatabaseTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactConverter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContact.java  4
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java  1
- app/src/test/java/ingsw/group1/findmyphone/contacts/SMSContactConverterTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954&bid=16307297)

RicDeZen commented on app/src/test/java/ingsw/group1/findmyphone/contacts/SMSContactConverterTest.java:
```diff
@@ -0,0 +1,59 @@
+    private static final String CONTACT_VALID_NAME = "NewContact";
+    private static final String CONTACT_EMPTY_NAME = "";
+
+    private SMSContact contactWithoutName, contactWithName; //a contact may or may not have a name
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954)

bortoletti-giorgia committed:
Changes after Codacy's suggestion

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 1
           

Complexity increasing per file
==============================
- app/src/test/java/ingsw/group1/findmyphone/contacts/SMSContactConverterTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContact.java  4
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactDatabaseTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/BaseDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactConverter.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactAdapter.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactManagerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954&bid=16307297)

bortoletti-giorgia committed:
Changed the class name from SMSContactConverter to SMSContactConverterUtils

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContact.java  4
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactManagerTest.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactDatabaseTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
- app/src/test/java/ingsw/group1/findmyphone/contacts/SMSContactConverterTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactConverterUtils.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/BaseDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactDao.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954&bid=16307297)

bortoletti-giorgia committed:
Resolved some conflicts with branch development

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/BaseDao.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactManagerTest.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactDatabaseTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactDao.java  1
- app/src/test/java/ingsw/group1/findmyphone/contacts/SMSContactConverterTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContact.java  4
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactConverterUtils.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954&bid=16307297)

bortoletti-giorgia committed:
Merge branch 'development' into feature/contacts

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  1
- app/src/test/java/ingsw/group1/findmyphone/contacts/SMSContactConverterTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ContactAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContact.java  4
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactDatabaseTest.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/BaseDao.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactConverterUtils.java  1
- app/src/androidTest/java/ingsw/group1/findmyphone/contacts/SMSContactManagerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4929954&bid=16307297)

ricdezen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContact.java:
```diff
@@ -0,0 +1,100 @@
+     * @return true if this object is equal to obj argument, false otherwise.
+     */
+    public boolean equals(@NonNull Object obj) {
+        if (obj == null || !(obj instanceof SMSContact))
```
Minuscolo consiglio last minute che non sei costretta a seguire perchè di fatto chissene frega, `instanceof` copre già il caso in cui `obj` sia `null`.

bortoletti-giorgia replied:
Terrei `== null` per leggibilità

ricdezen reviewed (approved):
All seems in order. I hope at least this time I didn't miss anything

ricdezen commented on app/src/test/java/ingsw/group1/findmyphone/contacts/SMSContactConverterTest.java:
```diff
@@ -0,0 +1,60 @@
+     */
+    @Test
+    public void peerFromContact() {
+        Assert.assertEquals(peerTest, SMSContactConverterUtils.peerFromContact(contactWithoutName));
```
Weird, Codacy shouldn't like the double assertion, whatever, not a huge deal.

bortoletti-giorgia replied:
What do you mean for assertion? 'Assert.assertEquals' ?

ricdezen replied:
Yes, usually tests should have a single assert/fail statement. You can always combine the two statements into a single `assertTrue` or `assertFalse`, and Codacy usually warns about this, but this time it didn't, no clue why

bortoletti-giorgia replied:
I don't know why
